### ISSUE TITLE
Podman support

### ISSUE THREAD

**Issue by Herohtar** [nhost/cli#727]
https://github.com/nhost/cli/issues/139 was closed with a comment saying that CLI 1.x now works with podman, but it does not appear to be true. I get the following output when running `nhost up`:
```
Setting up Nhost development environment...
Starting Nhost development environment...
Error: unknown flag: --project-directory
See 'podman --help'
failed to start Nhost development environment: failed to start docker compose: exit status 125
```

As I understand it, podman itself *can* work with `docker-compose`, but it appears that the CLI is running the `docker compose ...` command rather than `docker-compose`. That means it gets passed to podman via `podman-docker`, and the `compose` command is not supported by podman.

Indeed, if I run `docker compose --project-directory ...` I get the same output, which seems to confirm the cause of the CLI's error:
```
Error: unknown flag: --project-directory
See 'podman --help
```

**Comment by dbarrosop** [comment]
Could you try after install [podman-compose](https://github.com/containers/podman-compose)?

**Comment by dbarrosop** [comment]
Would you mind testing one of the artifacts from here? (scroll to the bottom)

https://github.com/nhost/cli/actions/runs/5287387814?pr=729

**Comment by Herohtar** [comment]
```
$ cli up
Setting up Nhost development environment...
Starting Nhost development environment...
podman-compose version: 1.0.6
usage: podman-compose [-h] [-v] [--in-pod in_pod] [--pod-args pod_args]
                      [--env-file env_file] [-f file] [-p PROJECT_NAME]
                      [--podman-path PODMAN_PATH] [--podman-args args]
                      [--podman-pull-args args] [--podman-push-args args]
                      [--podman-build-args args] [--podman-inspect-args args]
                      [--podman-run-args args] [--podman-start-args args]
                      [--podman-stop-args args] [--podman-rm-args args]
                      [--podman-volume-args args] [--no-ansi] [--no-cleanup]
                      [--dry-run]
                      {help,version,wait,systemd,pull,push,build,up,down,ps,run,exec,start,stop,restart,logs,config,port,pause,unpause,kill}
                      ...
podman-compose: error: argument command: invalid choice: '/nhost-test' (choose from 'help', 'version', 'wait', 'systemd', 'pull', 'push', 'build', 'up', 'down', 'ps', 'run', 'exec', 'start', 'stop', 'restart', 'logs', 'config', 'port', 'pause', 'unpause', 'kill')
failed to start Nhost development environment: failed to start docker compose: exit status 2
```

**Comment by dbarrosop** [comment]
Apologies for the late response. It is unfortunate that `podman-compose` isn't compatible with `docker-compose`. I guess the easiest thing to do would be to fallback to `docker-compose` with podman. Thoughts?

**Comment by Herohtar** [comment]
I believe Podman officially supports `docker-compose`, so going through that should be fine.

**Comment by awidjaja** [comment]
> I believe Podman officially supports `docker-compose`, so going through that should be fine.

Installing docker-compose doesn't solve the issue because it is already deprecated by docker. 
Docker expect a separate plugin (docker-cli-compose) installed and then it will be used by the `docker compose`. command.

The need here is to launch nhost containers locally using podman - so `nhost up` command should detect if podman-compose is installed and use it instead. A better option for podman is to also provide the kubernetes yaml format because podman can launch containers inside a pod using `podman play kube` command.

**Comment by dbarrosop** [comment]
> so nhost up command should detect if podman-compose is installed and use it instead.

The problem is that `podman-compose` isn't compatible with `docker-compose` so it requires an extra effort to handle. While this may be fine, this isn't a priority for us right now. If someone wants to contribute it that'd be great though.

**Comment by awidjaja** [comment]
May I know what argument was use in your test above for podman-compose? 

According to [this article](https://linuxhandbook.com/podman-compose/), podman-compose is compatible with docker-compose.yml without any change.

`podman-compose up -d -f .nhost/docker-compose.yml`

**Comment by Herohtar** [comment]
I think the issue is that the Docker command `compose` (`docker compose`, which is *not* `docker-compose`) has some flags that aren't supported by `podman-compose`. The test simply changed the command to run `podman-compose` without changing any of the arguments.

**Comment by dbarrosop** [comment]
> The test simply changed the command to run podman-compose without changing any of the arguments

Correct, it was just a dummy change to see if it would work. You can see it here + the needed arguments here:

https://github.com/nhost/cli/pull/729/files

**Comment by awidjaja** [comment]
Thanks both for clarifying. It's an issue with the `--project-directory`  cli parameter that is not used in podman-compose. 

According to [this](https://docs.docker.com/compose/reference/#:~:text=When%20you%20use%20multiple%20Compose%20files%2C%20all%20paths%20in%20the%20files%20are%20relative%20to%20the%20first%20configuration%20file%20specified%20with%20%2Df.%20You%20can%20use%20the%20%2D%2Dproject%2Ddirectory%20option%20to%20override%20this%20base%20path.) docker compose cli reference, that parameter is for specific case where multiple docker-compose.yml files that reside is different directories. In my understanding, nhost generate only a single docker-compose.yml file inside .nhost sub-directory.  

My suggestion on how to make it work for podman-compose: 

Use `docker version` command to check whether podman is used, and in that case use `podman-compose up` instead of `docker compose up` without the `--project-directory` argument , instead include the working directory in the `-f` argument. (in the test example, change from: `dc.filepath`  to `dc.workingDir+"/"+dc.filepath`
```bash
$ docker version
Client:       Podman Engine
Version:      4.5.1
API Version:  4.5.1
Go Version:   go1.18.10
Built:        Thu Jan  1 00:00:00 1970
OS/Arch:      linux/amd64

Server:       Podman Engine
Version:      4.5.0
API Version:  4.5.0
Go Version:   go1.19.7
Built:        Fri Apr 14 15:42:56 2023
OS/Arch:      linux/amd64

# or alternatively use `which podman-compose` to check if it's installed.

$ which podman-compose
/usr/local/bin/podman-compose

# Result of nhost up for docker compose
docker compose up -f ~/.nhost/docker-compose.yml

# Result of nhost up for podman
podman-compose up -f ~/.nhost/docker-compose.yml
```

**Comment by Nikhil-42** [comment]
Update: On Arch Linux (and Docker Desktop platforms) `docker-compose` redirects to the V2 `docker compose` engine. Podman will prefer to use `docker-compose` when available over `podman-compose`. So with this setup, the proper compose engine is invoked and the CLI runs (`docker compose` -> `podman compose` -> `docker-compose` -> Docker Compose V2).

However, there are issues running the traefik container. The generated `.nhost/docker-compose.yml` attempts to volume mount the `/var/run/docker.sock` which requires root. Similarly, the container tries to bind to :443 which also requires root. Ideally, root access shouldn't be necessary for running the compose, and the compose file should attempt to volume mount whatever the active socket is, not the root socket.
